api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/ClusterEndpointProvider.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.service;
+
+import java.util.Optional;
+
+/**
+ * Provides the addresses of the network endpoints required by a virtual cluster.
+ */
+public interface ClusterEndpointProvider {
+
+    /**
+     * Address of the cluster's bootstrap address.
+     *
+     * @return cluster's bootstrap address.
+     */
+    HostPort getClusterBootstrapAddress();
```

Comment:
More for my understanding than anything else, why are bootstrap addresses special in Kroxylicious? 

In my mental model they are just another localPort to upstream address mapping. The proxy has no special knowledge of bootstrap addresses (especially in the static topology model).

Reply:
I just see it as a reflection of kafka's own [bootstrap servers concept](https://kafka.apache.org/documentation/#producerconfigs_bootstrap.servers).  Clients need a bootstrap to connect the the cluster, and so kroxy does too.

The normal sequence of the events is that clients will connect to the virtual cluster's bootstrap address, which will connect to the clusters bootstrap address.  The client will then be in possession of the metadata, so it knows which broker it needs to connect to to perform its produce or fetch.  It connect to the broker endpoint exposed by kroxy, which in turn connects to the cluster's actual broker endpoint.


Yes, it is just another upstream mapping.